Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -433,20 +434,56 @@ public override void Update(float frameTime)
 
                 if (flammable.FireStacks > 0)
                 {
-                    var air = _atmosphereSystem.GetContainingMixture(uid);
+                    var air = _atmosphereSystem.GetContainingMixture(uid, false, true);
 
-                    // If we're in an oxygenless environment, put the fire out.
-                    if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                    if (flammable.FuelGasMix != null)
                     {
-                        Extinguish(uid, flammable);
-                        continue;
+                        // If we're in no atmos extinguish.
+                        if (air == null)
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
+
+                        // If our local atmos doesnt have enough of the required gases extinguish.
+                        for (int i = 0; i < Atmospherics.AdjustedNumberOfGases; i++)
+                        {
+                            if (flammable.FuelGasMix.GetMoles(i) > air.GetMoles(i))
+                            {
+                                Extinguish(uid, flammable);
+                                continue;
```

Comment:
If it extinguishes, it should break out of the loop instead of continue, right?

Reply:
Yes. I'll move it to a method on GasMixtures so I can avoid the nested looping.

Reply:
Done!